Cython/Utility/NumpyImportArray.c

```diff
@@ -0,0 +1,20 @@
+///////////////////////// NumpyImportArray.init ////////////////////
+
+// comment below is deliberately kept in the generated C file to
+// help users debug where this came from:
+/*
+ * Cython has automatically inserted a call to _import_array since
+ * you didn't include one when you cimported numpy. To disable this
+ * add the line
+ *   <void>_import_array
+ */
+#ifdef NPY_NDARRAYOBJECT_H /* numpy headers have been included */
+// NO_IMPORT_ARRAY is Numpy's mechanism for indicating that import_array is handled elsewhere
+#if !NO_IMPORT_ARRAY /* https://docs.scipy.org/doc/numpy-1.17.0/reference/c-api.array.html#c.NO_IMPORT_ARRAY  */
+if (_import_array() == -1) {
```

Comment:
```suggestion
if (unlikely(_import_array() == -1)) {
```

Reply:
I've added it but this but is called once in the module init code - it isn't that performance critical.

Reply:
Not at all, no. I was thinking that a) we use `unlikely()` (almost) wherever we do exception handling, so why not here, and b) it can have an impact on code layout. I don't want the C compiler to draw unlucky conclusions about the amount of space and memory pages to allocate to the module init function, so more hints about what will be used can't really hurt.